File: automation/java-selenium-ui/src/test/java/pages/ScrollbarsPage.java
package pages;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class ScrollbarsPage {
    private WebDriver driver;
    private By title = By.cssSelector("section .container h3");
    private By hiding_button = By.id("hidingButton");

    public ScrollbarsPage(WebDriver driver) {
        this.driver = driver;
    }

    public String getTitle(){
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.presenceOfElementLocated(title));
        return driver.getTitle();
    }

    public void clickHidingButton() {
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        WebElement hiddenButton = wait.until(ExpectedConditions.presenceOfElementLocated(hiding_button));
        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true)", hiddenButton);
        hiddenButton.click();
    }

}

File: automation/cypress/cypress/page-objects/scrollbars-page.js
 /// <reference types="cypress" />
 export class ScrollbarsPage {
    
    getTitle(){
        return cy.title('section .container h3');
    }
    
    clickHiddenButton(){
        cy.get('#hidingButton')
        .scrollIntoView()   // Лучше подходит для "спрятанных" элементов
        .should('be.visible')  // Убедиться, что элемент видим
        .click();
        
    }
 }